src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -75,15 +75,13 @@ class LibraryService (protected val argUserInfo: UserInfo,
 
   def updateDiscoverableByGroups(ns: String, name: String, newGroups: Seq[String]): Future[PerRequestMessage] = {
     if (newGroups.forall { g => FireCloudConfig.ElasticSearch.discoverGroupNames.contains(g) }) {
-      rawlsDAO.getWorkspace(ns, name) map { workspaceResponse =>
-        if (workspaceResponse.accessLevel >= WorkspaceAccessLevels.Owner) {
-          // this is technically vulnerable to a race condition in which the workspace attributes have changed
-          // between the time we retrieved them and here, where we update them.
-          val remove = Seq(RemoveAttribute(discoverableWSAttribute))
-          val operations = newGroups map (group => AddListMember(discoverableWSAttribute, new AttributeString(group)))
-          RequestComplete(patchWorkspace(ns, name, remove ++ operations, isPublished(workspaceResponse)))
-        } else {
-          RequestCompleteWithErrorReport(Forbidden, "must be an owner to set discoverable groups")
+      rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
+        val remove = Seq(RemoveAttribute(discoverableWSAttribute))
+        val operations = newGroups map (group => AddListMember(discoverableWSAttribute, new AttributeString(group)))
```

Comment:
Nit: Not sure if your IDE shows this, but you don't need the `new` keyword here.